File: src/main/java/com/group/educate/Model/Job/Internship.java
//Made By Eyad

package com.group.educate.Model.Job;

import com.group.educate.Model.User.Company.Company;

public class Internship extends JobPosting{
    private int duration;
    private Company company;

    public Internship(String JobDescription, String JobName, String JobRequirements, String JobTitle, PostingType jobPostingType,  int duration,Company company) {
        super(JobDescription, JobName, JobRequirements, JobTitle, jobPostingType);
        this.duration = duration;
        this.company=company;
    }

    public Internship(String internshipID, String jobDescription, String jobName, String jobRequirements, String jobTitle, PostingType jobPostingType, int duration, Company company) {
        super(internshipID, jobDescription, jobName, jobRequirements, jobTitle, jobPostingType);
        this.duration = duration;
        this.company=company;
    }

    public int getDuration() {
        return duration;
    }

    @Override
    public String toString() {
        return super.toString()
                + duration + '|'
                + company.toString() + '|';
    }
}

File: Frontend/app/Model/Jobs/Internship.ts
import type { Company } from "../Company";
import type { Recruiter } from "../Users/Recruiter";
import { JobPosting } from "./JobPosting";

class Internship extends JobPosting {

    duration: number;
    jobLocation: String;

    constructor(id: bigint, jobDescription: String, jobRequirements: String, jobName: String, company: Company, jobLocation: String, postingDate: Date, recruiter: Recruiter, duration: number) {
        super(id, jobDescription, jobRequirements, jobName, company, jobLocation, postingDate, recruiter, "Internship");
        this.duration = duration;
        this.jobLocation = jobLocation;
    }
}